src/bindings.ts

```diff
@@ -47,12 +62,25 @@ export function bindRoutes(routerRoutes: any, controllers: any[], getter?: (ctrl
   for(const ctrl of controllers) {
     const routes = Reflect.getMetadata(ROUTE_PREFIX, ctrl);
 
-    for(const { method, url, middleware, name, params } of routes) {
-      routerRoutes[method](url, ...middleware, async function(ctx, next) {
+    for(const { method, url, middleware, name, params, asyncParams } of routes) {
+      routerRoutes[method](url, ...middleware, async (ctx, next) => {
         const inst = getter === undefined ?
           new ctrl() : getter(ctrl);
 
-        const args = getArguments(params, ctx, next);
+        let args = [];
+
+        if (params) {
+          callArguments(params, args, ctx);
+        }
+        
+        if (asyncParams) {
+          await callAsyncArguments(asyncParams, args, ctx);
+        }
```

Comment:
I'm not a fan of functions that modify arguments instead of returning.
Maybe modify the current `getArguments()` to be async and accept both `params` and `asyncParams`.

Reply:
I don't want to create a promise by awaiting getArguments() when there is no reason to await for something. Please check my new commit, getArguments() and getAsyncArguments() are now returning the args array, no more modifying inside the functions